File: packages/react-native/ios/DashXLog.swift
import Foundation
import os.log

class DashXLog {
    enum LogLevel: Int {
        case info = 1
        case debug = 0
        case off = -1
                
        static func <= (lhs: Self, rhs: Self) -> Bool {
            return lhs.rawValue <= rhs.rawValue
        }
        
        static func > (lhs: DashXLog.LogLevel, rhs: DashXLog.LogLevel) -> Bool {
            return lhs.rawValue > rhs.rawValue
        }
    }
    
    private var logLevel: LogLevel = .off
    
    func setLogLevel(to: Int) {
        self.logLevel = LogLevel(rawValue: to) ?? .off
    }
    
    func d(tag: String, data: String) {
        if (logLevel <= .debug && logLevel > .off) {
            if #available(iOS 10.0, *) {
                os_log("%@: %@", type: .debug, tag, data)
            } else {
                print("%@: %@", tag, data)
            }
        }
    }
    
    func i(tag: String, data: String) {
        if (logLevel <= .info && logLevel > .off) {
            if #available(iOS 10.0, *) {
                os_log("%@: %@", type: .info, tag, data)
            } else {
                print("%@: %@", tag, data)
            }
        }
    }
}

let Logger = DashXLog()

File: packages/react-native/android/src/main/java/com/dashx/sdk/DashXLog.kt
package com.dashx.sdk

import android.util.Log

object DashXLog {
    private val tag = DashXLog::class.java.simpleName

    private enum class LogLevel(val code: Int) {
        INFO(1), DEBUG(0), OFF(-1);

        companion object {
            fun lookup(code: Int): LogLevel {
                return when (code) {
                    1 -> INFO
                    0 -> DEBUG
                    else -> OFF
                }
            }
        }
    }

    private var logLevel = LogLevel.OFF

    @JvmStatic
    fun setLogLevel(logLevel: Int) {
        DashXLog.logLevel = LogLevel.lookup(logLevel)
    }

    @JvmStatic
    fun d(tag: String? = this.tag, logText: String?) {
        if (logLevel.code >= LogLevel.DEBUG.code) {
            Log.d(tag, logText ?: "")
        }
    }

    @JvmStatic
    fun i(tag: String? = this.tag, logText: String?) {
        if (logLevel.code >= LogLevel.INFO.code) {
            Log.i(tag, logText ?: "")
        }
    }
}
